### PR TITLE
Crategroups

### PR DESCRIPTION
**Pull request by Bhaskar70**

### CODE REVIEW

**Review by mahesh818480 — APPROVED** [review]

File: src/app/components/chat-box/chat-box.component.ts [hunk 1]
```diff
@@ -39,24 +32,33 @@ export class ChatBoxComponent {
...
+      data: {userId : this.currentUser._id},
       key: 'newUser',
     });
-    this.chatservice.getSocketData('newUser').subscribe((res) => {
-      this.UserListData = res;
+    this.chatservice.getSocketData('newUser').subscribe(({ userPayload, userId, opponentPayload, opponentId}) => {
+     if(this.currentUser._id == userId){
+       this.UserListData = this.currentUser.isAdmin ? userPayload : this.getOnlyAdmins(userPayload);
+     }else if(this.currentUser._id == opponentId){
+      this.UserListData =this.currentUser.isAdmin ? opponentPayload : this.getOnlyAdmins(userPayload);
```
Comment (ResourceOne2011) [inline src/app/components/chat-box/chat-box.component.ts:58]: change userPayload to adminPayload in the 58th line

File: src/app/components/chat-box/chat-box.component.ts [hunk 2]
```diff
@@ -67,12 +69,21 @@ export class ChatBoxComponent {
       //  this.route.navigate([''])
     }
     this.UserSelected = 'Test';
+    this.chatservice.getSocketData('groupCreated').subscribe((res) => {
```
Comment (ResourceOne2011) [inline src/app/components/chat-box/chat-box.component.ts:72]: Push newly created group to the existed groups

File: src/app/services/chat.service.ts [hunk 3]
```diff
@@ -183,6 +183,13 @@ export class ChatService {
       };
     });
   }
+
+  // groups 
+
+  getAllGroups(id:any){
```
Comment (ResourceOne2011) [inline src/app/services/chat.service.ts:189]: for admin we need all groups , but for user we need only user groups, for admin call "/groups" only dont pass id

File: src/app/components/chat-box/chat-box.component.ts (hunk 2 above)
Comment (Bhaskar70) [reply src/app/components/chat-box/chat-box.component.ts:72]: done

File: src/app/components/chat-box/chat-box.component.ts (hunk 1 above)
Comment (Bhaskar70) [reply src/app/components/chat-box/chat-box.component.ts:58]: done

File: src/app/services/chat.service.ts (hunk 3 above)
Comment (Bhaskar70) [reply src/app/services/chat.service.ts:189]: done

**Pull request merged**